python/sqlflow_submitter/xgboost/train.py

```diff
@@ -153,16 +163,23 @@ def train(datasource,
                         evals_result=re,
                         xgb_model=bst,
                         **train_params)
-        bst.save_model("my_model")
         print("Evaluation result: %s" % re)
-    if is_pai and rank == 0 and len(oss_model_dir) > 0:
-        save_model(oss_model_dir, model_params, train_params, feature_metas,
-                   feature_column_names, label_meta)
+
+    if rank == 0:
+        # TODO(sneaxiy): save pmml as well
+        bst.save_model("my_model")
+
+        if is_pai and len(oss_model_dir) > 0:
+            save_model(oss_model_dir, model_params, train_params,
+                       feature_metas, feature_column_names, label_meta,
+                       feature_column_code)
 
 
 def save_model(model_dir, model_params, train_params, feature_metas,
-               feature_column_names, label_meta):
-    model.save_file(model_dir, "my_model")
+               feature_column_names, label_meta, feature_column_code):
+    model_name = "my_model"
```

Comment:
Why do we hard code the model name is `my_model`?

Reply:
String literal "my_model" is hard-coded in many places, for example [1](https://github.com/sql-machine-learning/sqlflow/blob/develop/pkg/sql/codegen/pai/template_xgboost.go#L38). I guess the key problem here is that what the name of the saved model is. Any name is OK but we should not hard code them in many places. We should clean them all in the near future.